dev/ProjectReunion_BootstrapDLL/MddBootstrap.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddBootstrap.h"
+
+#include "IDynamicDependencyLifetimeManager.h"
+
+STDAPI AddFrameworkToPath(PCWSTR  path);
+
+IDynamicDependencyLifetimeManager* g_lifetimeManager = nullptr;
+DLL_DIRECTORY_COOKIE g_dllDirectoryCookie = 0;
+wil::unique_cotaskmem_string g_frameworkPath;
+
+STDAPI MddBootstrapInitialize(
+    const CLSID& appDynamicDependencyLifetimeManager)
+{
+    wil::com_ptr_nothrow<IDynamicDependencyLifetimeManager> lifetimeManager;
+    RETURN_IF_FAILED(CoCreateInstance(appDynamicDependencyLifetimeManager, nullptr, CLSCTX_LOCAL_SERVER, __uuidof(IDynamicDependencyLifetimeManager), reinterpret_cast<void**>(lifetimeManager.addressof())));
+
+    wil::unique_cotaskmem_string path;
+    RETURN_IF_FAILED(lifetimeManager->GetFrameworkPath(path.addressof()));
+
+    RETURN_IF_FAILED(AddFrameworkToPath(path.get()));
+
+    g_lifetimeManager = lifetimeManager.detach();
+    return S_OK;
+}
+
+STDAPI MddBootstrapShutdown()
+{
+    if (!g_lifetimeManager)
+    {
+        RETURN_IF_FAILED(g_lifetimeManager->Shutdown());
+        g_lifetimeManager->Release();
+        g_lifetimeManager = nullptr;
+    }
+
+    if (g_dllDirectoryCookie != 0)
+    {
+        RemoveDllDirectory(g_dllDirectoryCookie);
+        g_dllDirectoryCookie = 0;
+    }
+
+    if (g_frameworkPath)
+    {
+        wil::unique_cotaskmem_string path;
+        const auto hr = wil::TryGetEnvironmentVariableW(L"PATH", path);
+        if (SUCCEEDED(hr) && path)
+        {
+            const auto frameworkPathLength = wcslen(g_frameworkPath.get());
+            PCWSTR pathWithoutFrameworkPath = g_frameworkPath.get() + frameworkPathLength;
+            if (*pathWithoutFrameworkPath == L';')
+            {
+                ++pathWithoutFrameworkPath;
+            }
+            (void)LOG_IF_WIN32_BOOL_FALSE(SetEnvironmentVariableW(L"PATH", pathWithoutFrameworkPath));
+        }
+    }
+
+    return S_OK;
+}
+
+STDAPI AddFrameworkToPath(PCWSTR frameworkPath)
+{
+    // Add the framework to the Loader's DllDirectory list
+    auto dllDirectoryCookie = AddDllDirectory(frameworkPath);
+    RETURN_HR_IF(HRESULT_FROM_WIN32(GetLastError()), dllDirectoryCookie == 0);
```

Comment:
`RETURN_LAST_ERROR_IF(dllDirectoryCookie)` ?

Reply:
RETURN_LAST_ERROR_IF +1

(dllDirectoryCookie) -1
"Wrong Type: bool/BOOL/BOOLEAN/boolean expected"

RETURN_LAST_ERROR_IF(dllDirectoryCookie == 0) is needed